api/src/main/java/com/qaengine/controllers/QuestionController.java

```diff
@@ -1,51 +1,66 @@
 package com.qaengine.controllers;
 
-import com.qaengine.database.AnswerRepository;
 import com.qaengine.database.QuestionRepository;
 import com.qaengine.exceptions.ResourceNotFoundException;
 import com.qaengine.lib.HelperFunctions;
+import com.qaengine.models.Category;
 import com.qaengine.models.Question;
 import com.qaengine.models.inputs.QuestionInput;
+import com.qaengine.models.inputs.QuestionListInput;
+import com.qaengine.models.outputs.QuestionList;
 import com.qaengine.models.outputs.QuestionListElement;
-import com.qaengine.services.AnswerService;
+import com.qaengine.services.CategoryService;
 import com.qaengine.services.QuestionService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.domain.PageRequest;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.data.domain.Page;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
 
 import javax.validation.Valid;
 import java.util.List;
 
 @RestController
 public class QuestionController {
+    private QuestionRepository questionRepository;
+    private QuestionService questionService;
+    private CategoryService categoryService;
+
     @Autowired
-    QuestionRepository questionRepository;
-    @Autowired
-    AnswerRepository answerRepository;
-    @Autowired
-    AnswerService answerService;
-    @Autowired
-    QuestionService questionService;
+    public QuestionController(QuestionRepository questionRepository, QuestionService questionService, CategoryService categoryService) {
+        this.questionRepository = questionRepository;
+        this.questionService = questionService;
+        this.categoryService = categoryService;
+    }
 
     @GetMapping("/questions")
     @CrossOrigin()
-    protected List<QuestionListElement> listQuestions(
-            @RequestParam(value = "page") Integer page,
-            @RequestParam(value = "limit") Integer limit) {
-        return questionService.listQuestions(page, limit);
+    protected QuestionList listQuestions(QuestionListInput input) {
```

Comment:
Are the QuestionListInput's attributes `RequestParam`s? Like `GET site.com?page=1&limit=2`? Did you test it? If so, I didn't know you could do that, cool.

Reply:
Yes, they are. It's similar to POST requests @RequestBody.
Tested ofcourse.